vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.jsp

```diff
@@ -89,96 +106,222 @@ $(document).ready(function() {
 	});
 	$("#addImgAlert").draggable();
 	$("#addTextAlert").draggable();
+	
+	// ------------- edit text block ----------------
+	function onMouseEnter(e){
+		var target = $( e.target );
+        $(".hova").removeClass("hova");
+        $(e.target).addClass("hova");
+        // This is needed to prevent only the p tag from being highlighted
+        if(target.is('p')){
+            $(e.target).parent().addClass("hova");
+        }
+        //prevent normal hover actions
+        e.preventDefault();
+    }
+
+    function onMouseLeave(e) {
+        $(this).removeClass("hova");
+    }
 
+    function onDoubleClick(e){
+        // get the id of the nearest div with id valueDiv
+        var blockId = $(e.target).closest('div').attr('id');
+        // if there is a block id we know its text block otherwise its an image block
+        if(blockId){
+        	//remove card border
+            $("#" + blockId).css('border', 'none');
+        	//set id in local storage so it can be retrieved in other functions
+            localStorage.setItem('textBlockId', blockId)
+            // get text from p tag
+            var description = $("#" + blockId + " p").text();
+            // insert text box and buttons
+            $('<div class="col-xs-12" id="newTextBlockDiv" ><textarea id="newTextBlock" style="margin-top: 1%;" class="form-control" type="text">'+description+'</textarea></div>').insertBefore( "#" + blockId + " p" );
+            $('<div class="col-xs-1" style="margin-top: 1%"><a id="cancelTextBlock" class="btn" href="#"style="float: right;"><i class="fas fa-times"></i></a><a id="submitTextBlock" class="btn" href="#"style="float: right;"><i class="fas fa-check"></i></a></div>').insertAfter( "#newTextBlockDiv" );
+            $("#" + blockId + " p").remove();
+            // unbind events to prevent multiple instances of buttons and constant highlighting
+            $('.valueDiv').unbind('mouseenter mouseleave dblclick');
+            // remove highlighting if present
+            $(this).removeClass("hova");
+        } else {
+        	// Must use closest('div').children() instead of closet('img').
+        	blockId = $(e.target).closest('div').children().attr('id');
+        	console.log(blockId);
+            $("#addImgAlert").show();
+            localStorage.setItem('imgBlockId', blockId);
```

Comment:
is there a reason you not just adding the block id as attribute to the submit button or alert? using local storage seems a little overkill and has higher bugginess potential.

Reply:
I use the block ids in order to keep track of which block I am appending and deleting UI components from as well as in the form submission. I can remove it from the submissions but it will still be used else where to manipulate the UI. Let me know what you would prefer.

Reply:
I would reevaluate if you really need local storage. You can probably get most things done without using local storage. What exactly are you keeping track of? can't you just add ids to the elements and use them when clicking on buttons to delete something?